src/core/utils/stringutils.h

```diff
@@ -0,0 +1,46 @@
+/***************************************************************************
+                        stringutils.h
+                        ---------------
+  begin                : Jun 2020
+  copyright            : (C) 2020 by Ivan Ivanov
+  email                : ivan@opengis.ch
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef STRINGUTILS_H
+#define STRINGUTILS_H
+
+#include <QObject>
+
+
+class StringUtils : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+
+    explicit StringUtils( QObject *parent = nullptr );
+
+
+    /**
+     * Checks whether the provided string is a relative url (has no protocol or starts with `file://`).
+     */
+    static Q_INVOKABLE bool isRelativeUrl( const QString &url );
```

Comment:
what about moving this to existing file utils?

Reply:
Is it really a file util or a string util? At least for me it feels weird to have check for strings that start with `http://` in `FileUtils`. No strong opinions though.

Reply:
I also think there would be a better place than generic string utils, the strings we deal with here already have url semantic.
I'd move it either to FileUtils or UrlUtils, given that `http://` really isn't very file specific.

Reply:
Added new `UrlUtils`, feels better than `FileUtils`